src/gamestates/cMainMenuGameState.cpp

```diff
@@ -63,81 +61,113 @@ cMainMenuGameState::cMainMenuGameState(cGame &theGame) : cGameState(theGame), te
     // Buttons:
     int buttonsX = mainMenuFrameX + 2;
 
-    Color colorInactiveHover = Color{128, 128, 128,255};
-    Color colorInactive = Color{192, 192, 192,255};
-
     // PLAY
     int playY = 323;
 
     buttonWidth = mainMenuWidth - 8;
 
     const cRectangle &window = cRectangle(mainMenuFrameX, mainMenuFrameY, mainMenuWidth, mainMenuHeight);
-    gui_window = new cGuiWindow(window);
+    gui_window = std::make_unique<GuiWindow>(window);
+    gui_window->setTheme(GuiTheme::Light());
 
     const cRectangle &campaign = cRectangle(buttonsX, playY, buttonWidth, buttonHeight);
-
-    cGuiButton *gui_btn_SelectHouse = new cGuiButton(textDrawer, campaign, "Campaign", buttonKinds);
-    gui_btn_SelectHouse->setTextAlignHorizontal(buttonTextAlignment);
-    gui_btn_SelectHouse->setOnLeftMouseButtonClickedAction(new cGuiActionSelectHouse());
-    gui_window->addGuiObject(gui_btn_SelectHouse);
+    std::unique_ptr<GuiButton> gui_btn_SelectHouse = GuiButtonBuilder()
+            .withRect(campaign)        
+            .withLabel("Campaign")
+            .withTextDrawer(&textDrawer)    
+            .withTheme(GuiTheme::Light())
+            .withKind(GuiRenderKind::TRANSPARENT_WITHOUT_BORDER)
+            .onClick([this]() {
+                game.setNextStateToTransitionTo(GAME_SELECT_HOUSE);
+                game.initiateFadingOut();})
+            .build();
+    gui_window->addGuiObject(std::move(gui_btn_SelectHouse));
 
     int skirmishY = 344;
     const cRectangle &skirmish = cRectangle(buttonsX, skirmishY, buttonWidth, buttonHeight);
-    cGuiButton *gui_btn_Skirmish = new cGuiButton(textDrawer, skirmish, "Skirmish", buttonKinds);
-    gui_btn_Skirmish->setTextAlignHorizontal(buttonTextAlignment);
-    gui_btn_Skirmish->setOnLeftMouseButtonClickedAction(new cGuiActionSetupSkirmishGame());
-    gui_window->addGuiObject(gui_btn_Skirmish);
+    std::unique_ptr<GuiButton> gui_btn_Skirmish = GuiButtonBuilder()
```

Comment:
not sure if this is possible, but in my other projects I am used to this kind of reference to the Builder:

`ThingToBuild.builder().withX().withY().build()`. 

Could you change it so that this line reads like:
```suggestion
    std::unique_ptr<GuiButton> gui_btn_Skirmish = GuiButton.builder()
```
?